src/ui.jl

```diff
@@ -232,22 +232,44 @@ function HeaderREPLs.activate_header(header::RebugHeader, p, s, termbuf, term)
     set_uuid!(header, str)
 end
 
-## Key bindings
+const keybindings = Dict{Symbol,String}(
+    :stacktrace => "\e[15~",  # F5
+    :stepin => "\e\eOM",  # Alt-Shift-Enter
+    :deprecated_stepin => "\e[23~",  # F11
+)
 
-# For adding keys to already-constructed keymaps
+const modeswitches = Dict{Any,Any}(
+    :stacktrace => (s, o...) -> capture_stacktrace(s),
+    :stepin => (s, o...) -> (stepin(s); enter_rebug(s)),
+    :deprecated_stepin => (s, o...) -> (stepin(s); rebug_prompt_ref[].repl.header.warnmsg="stepin is now Alt-Shift-Enter"; enter_rebug(s)),
+)
 
-add_key_stacktrace!(keymap) = LineEdit.add_nested_key!(keymap, "\e[15~", (s, o...) -> capture_stacktrace(s))
-add_key_stepin!(keymap)     = LineEdit.add_nested_key!(keymap, "\e\eOM", (s, o...) -> (stepin(s); enter_rebug(s)))
+function get_rebugger_modeswitch_dict()
+    rebugger_modeswitch = Dict()
+    for (action, keybinding) in keybindings
+        rebugger_modeswitch[keybinding] = modeswitches[action]
+    end
+    @show rebugger_modeswitch
+end
 
-# These work at the `julia>` prompt and the `rebug>` prompt
-const rebugger_modeswitch = Dict{Any,Any}(
-    # F5
-    "\e[15~"   => (s, o...) -> capture_stacktrace(s),
-    # Alt-Shift-Enter
-    "\e\eOM"   => (s, o...) -> (stepin(s); enter_rebug(s)),
-    # Deprecate F11
-    "\e[23~"   => (s, o...) -> (stepin(s); rebug_prompt_ref[].repl.header.warnmsg="stepin is now Alt-Shift-Enter"; enter_rebug(s)),
-)
+function add_keybinding(action::Symbol, keybinding::String)
+    main_repl = Base.active_repl
+    repl = HeaderREPL(main_repl, RebugHeader())
```

Comment:
Won't this create a new REPL with every call to set a new key binding? So if you set two keys, won't it throw away the previous one? My prediction is that this will break `rebug>` mode but not `julia>` mode.